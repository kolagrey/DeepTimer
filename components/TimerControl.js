import React from 'react';
import { View } from 'react-native';
import { FAB } from 'react-native-paper';
import PropTypes from 'prop-types';
import styles from '../styles';

const TimerControl = (props) => {
    const { paused, play, pause } = props;
    return (
        <View style={styles.timerControlArea}>
            <FAB
                style={styles.fab}
                color="black"
                icon={paused ? 'play' : 'pause'}
                onPress={paused ? () => play() : () => pause()}
            />
        </View>
    );
};

TimerControl.propTypes = {
    paused: PropTypes.bool,
    play: PropTypes.func,
    pause: PropTypes.func
};

export default TimerControl;