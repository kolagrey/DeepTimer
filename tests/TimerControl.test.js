import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import TimerControl from '../components/TimerControl';


describe('<TimerControl/>', () => {

    test('renders correctly', () => {
        const tree = renderer.create(<TimerControl/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("is of type 'View'", () => {
        const tree = renderer
            .create(
                <TimerControl />,
            ).toJSON();
        expect(tree.type).toBe('View');
    });

    it('can hold a child', () => {
        const tree = renderer
            .create(
                <TimerControl>
                    <Text>
                        Hello Test
                    </Text>
                </TimerControl>,
            ).toJSON();
        expect(tree.children.length).toBe(1);
    });

    it('has a child View (FAB)', () => {
        const tree = renderer
            .create(
                <TimerControl/>,
            ).toJSON();
        expect(tree.children[0].type).toBe('View');
    });
});