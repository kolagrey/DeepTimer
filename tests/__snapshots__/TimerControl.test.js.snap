// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimerControl/> renders correctly 1`] = `
<View
  style={
    Object {
      "margin": 5,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 28,
        "elevation": 6,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 25,
        "opacity": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 6,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 28,
          },
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "height": 56,
              "width": 56,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
